inc/functions.php

```diff
@@ -357,8 +357,8 @@ function my_date($format, $stamp="", $offset="", $ty=1, $adodb=false)
 			$dstcorrection = $mybb->settings['dstcorrection'];
 		}
 
-		// If DST correction is enabled, add an additional hour to the timezone.
-		if($dstcorrection == 1)
+		// If DST correction is enabled and the date falls into DST, add an additional hour to the timezone.
+		if($dstcorrection == 1 || ($dstcorrection == 2 && date('I', $stamp) == 1))
```

Comment:
Since there are several different rules regarding daylight saving time (and countries not having it at all), it don't thing date('I', $stamp) is reliable.